util/types.c

```diff
@@ -45,16 +45,19 @@ uint64_t int48_to_long(__u8 *data)
 	return result;
 }
 
-char *uint128_t_to_string(nvme_uint128_t val)
+char *uint128_t_to_string_tg(nvme_uint128_t val, bool thousands_grouping)
 {
-	static char str[40];
-	int idx = 40;
+	static char str[60];
+	int idx = 60;
 	__u64 div, rem;
 
 	/* terminate at the end, and build up from the ones */
 	str[--idx] = '\0';
 
 	do {
+		if (thousands_grouping && !((sizeof(str) - idx) % 4))
+			str[--idx] = ',';
```

Comment:
Just rely on printf localization and use `"'%d"` 

```
The five flag characters above are defined in the C99 standard.  The Single UNIX Specification spec-
       ifies one further flag character.

       '      For  decimal  conversion  (i,  d,  u, f, F, g, G) the output is to be grouped with thousands'
              grouping characters if the locale information indicates any.  (See setlocale(3).)  Note  that
              many  versions  of  gcc(1) cannot parse this option and will issue a warning.  (SUSv2 did not
              include %'F, but SUSv3 added it.)
```

See also https://stackoverflow.com/questions/10602729/digit-grouping-in-cs-printf

Reply:
Yes but the variable type nvme_uint128_t is not used by printf I think. Instead of this the 128 bit value can be converted t o long doulbe then we can use printf to group the value with thousands' grouping characters.

Reply:
Ah yes, you are right. There is not support from glibc for this type. But we still can't just use a comma as grouping blindly as this is a localization question.  So I I think you should use `localeconv()` ( to retrieve the correct char for grouping and do that on default. This also avoids the bool argument.

Reply:
Okay I will do use `localeconv()` mentioned. But about the bool argument it is to keep the existing `uint128_t_to_string()` it does not use a comma as grouping and seems that it should be not changed.

Reply:
Indeed, if we do localization (which includes thousends grouping) we have to do it everywhere.